scipy/cluster/tests/test_vq.py

```diff
@@ -270,37 +323,44 @@ def test_krandinit(self):
             else:
                 rng = np.random.RandomState(1234)
 
-            init = _krandinit(data, k, rng)
-            orig_cov = np.cov(data, rowvar=0)
-            init_cov = np.cov(init, rowvar=0)
+            init = _krandinit(data, k, rng, xp)
+            orig_cov = xp.cov(data.T)
+            init_cov = xp.cov(init.T)
             assert_allclose(orig_cov, init_cov, atol=1e-2)
 
-    def test_kmeans2_empty(self):
+    @array_api_compatible
+    def test_kmeans2_empty(self, xp):
         # Regression test for gh-1032.
-        assert_raises(ValueError, kmeans2, [], 2)
+        assert_raises(ValueError, kmeans2, xp.asarray([]), xp.asarray(2))
 
+    @skip_if_array_api
     def test_kmeans_0k(self):
         # Regression test for gh-1073: fail when k arg is 0.
         assert_raises(ValueError, kmeans, X, 0)
         assert_raises(ValueError, kmeans2, X, 0)
         assert_raises(ValueError, kmeans2, X, np.array([]))
 
-    def test_kmeans_large_thres(self):
+    @array_api_compatible
+    def test_kmeans_large_thres(self, xp):
         # Regression test for gh-1774
-        x = np.array([1, 2, 3, 4, 10], dtype=float)
-        res = kmeans(x, 1, thresh=1e16)
-        assert_allclose(res[0], np.array([4.]))
+        x = xp.asarray([1, 2, 3, 4, 10], dtype=float)
```

Comment:
minor: while we're at it, let's clean up `dtype=float`? Use `xp.float64` instead.